dotnet-authserver/tests/TeacherIdentity.AuthServer.Tests/EndpointTests/SignIn/Trn/NiNumberTests.cs

```diff
@@ -0,0 +1,183 @@
+    }
+
+    [Fact]
+    public async Task Get_EmailNotVerified_RedirectsToEmailConfirmationPage()
+    {
+        // Arrange
+        var authStateHelper = await CreateAuthenticationStateHelper(c => c.EmailSet());
+
+        var request = new HttpRequestMessage(HttpMethod.Get, $"/sign-in/trn/have-nino?{authStateHelper.ToQueryParam()}");
+
+        // Act
+        var response = await HttpClient.SendAsync(request);
+
+        // Assert
+        Assert.Equal(StatusCodes.Status302Found, (int)response.StatusCode);
+        Assert.StartsWith("/sign-in/email-confirmation", response.Headers.Location?.OriginalString);
+    }
+
+    [Fact]
+    public async Task Get_OfficialNameNotSet_RedirectsToNextHopUrl()
+    {
+        // Arrange
+        var authStateHelper = await CreateAuthenticationStateHelper(c => c.EmailVerified());
+
+        var request = new HttpRequestMessage(HttpMethod.Get, $"/sign-in/trn/have-nino?{authStateHelper.ToQueryParam()}");
+
+        // Act
+        var response = await HttpClient.SendAsync(request);
+
+        // Assert
+        Assert.Equal(StatusCodes.Status302Found, (int)response.StatusCode);
+        Assert.Equal(authStateHelper.GetNextHopUrl(), response.Headers.Location?.OriginalString);
+    }
+
+    [Fact]
+    public async Task Get_ValidRequest_RendersContent()
+    {
+        await ValidRequest_RendersContent("/sign-in/trn/ni-number", c => c.OfficialNameSet());
+    }
+
+    [Fact]
+    public async Task Post_EmptyNiNumber_ReturnsError()
+    {
+        // Arrange
+        var authStateHelper = await CreateAuthenticationStateHelper(c => c.OfficialNameSet());
+
+        HostFixture.RateLimitStore.Setup(x => x.IsClientIpBlockedForPinGeneration(TestRequestClientIpProvider.ClientIpAddress)).ReturnsAsync(true);
+
+        var request = new HttpRequestMessage(HttpMethod.Post, $"/sign-in/trn/ni-number?{authStateHelper.ToQueryParam()}")
+        {
+            Content = new FormUrlEncodedContentBuilder()
+        };
+
+        // Act
+        var response = await HttpClient.SendAsync(request);
+
+        // Assert
+        await AssertEx.HtmlResponseHasError(response, "NiNumber", "Enter a National Insurance number");
+    }
+
+    [Theory]
+    [InlineData("QQ 123456C")]
+    [InlineData("123456")]
+    [InlineData("QQ12345C")]
+    public async Task Post_InvalidNiNumber_ReturnsError(string niNumber)
+    {
+        // Arrange
+        var authStateHelper = await CreateAuthenticationStateHelper(c => c.OfficialNameSet());
+
+        HostFixture.RateLimitStore.Setup(x => x.IsClientIpBlockedForPinGeneration(TestRequestClientIpProvider.ClientIpAddress)).ReturnsAsync(true);
+
+        var request = new HttpRequestMessage(HttpMethod.Post, $"/sign-in/trn/ni-number?{authStateHelper.ToQueryParam()}")
+        {
+            Content = new FormUrlEncodedContentBuilder()
+            {
+                { "NiNumber", niNumber },
+                { "submit", "submit" },
```

Comment:
possible to avoid having to add this to the POST request in tests if explicitly bind submit in PageModel, but I don't think that's the right approach